h1st/core/ensemble/stack_ensemble.py

```diff
@@ -29,15 +29,33 @@ def _preprocess(self, X: Any) -> Any:
         Predicts all sub models then merge input raw data with predictions for ensembler's training or prediction
         :param X: raw input data
         """
+        if isinstance(X, list):
+            predictions = []
+            for x in X:
+                pred_temp = []
+                for m in self._sub_models:
```

Comment:
line#37-40 and line#46-49 are identical, please extract into a function @TgithubJ